lib/protocol/protocol.go

```diff
@@ -388,7 +388,9 @@ func (c *rawConnection) dispatcherLoop() (err error) {
 			if state != stateInitial {
 				return fmt.Errorf("protocol error: cluster config message in state %d", state)
 			}
-			c.receiver.ClusterConfig(c.id, *msg)
+			if err := c.receiver.ClusterConfig(c.id, *msg); err != nil {
+				return fmt.Errorf("receiver error: %v", err)
```

Comment:
Please use errors.Wrap (don't think it matters here, but still). Same below

Reply:
Wrap'd